enterprise/server/cmd/goinit/main.go

```diff
@@ -184,6 +186,20 @@ func startDockerd(ctx context.Context) error {
 		return err
 	}
 
+	mirror, err := fetchFromMMDS("firecracker_vm_docker_mirror")
+	if err != nil {
+		return err
+	}
+	insecureRegistry, err := fetchFromMMDS("firecracker_vm_insecure_registry")
+	if err != nil {
+		return err
+	}
+	if mirror != "" || insecureRegistry != "" {
+		if err := writeDockerConfig(mirror, insecureRegistry); err != nil {
+			return err
+		}
+	}
```

Comment:
nit: I think it'd be a little more flexible/simpler to write the whole config json to a metadata key like `"docker_config_json"`, and also if `fetchFromMMDS` returns `[]byte` instead of `string` then we can `os.WriteFile()` with the bytes directly instead of using an intermediate string conversion.

Reply:
Not a nit!

I took this route, but decided to _always_ pass the configuration JSON under the `dockerd_daemon_json` key. Hopefully it does not impact boot time.